pytorch_lightning/callbacks/device_stats_monitor.py

```diff
@@ -80,7 +80,7 @@ def on_train_batch_end(
         if not trainer.loggers:
             raise MisconfigurationException("Cannot use `DeviceStatsMonitor` callback with `Trainer(logger=False)`.")
 
-        if not trainer.logger_connector.should_update_logs:
+        if not trainer._logger_connector.should_update_logs:
```

Comment:
n00b question: why does the callback need to check this vs simply calling logger.log_metrics?

Reply:
to follow the `log_every_n_steps` criteria since this is not controlled by loggers.